Content.Server/Atmos/Piping/Unary/Components/GasVentPumpComponent.cs

```diff
@@ -59,6 +59,48 @@ public sealed partial class GasVentPumpComponent : Component
         [DataField("underPressureLockoutLeaking")]
         public float UnderPressureLockoutLeaking = 0.0001f;
 
+        #region some averaging stuff
+        [ViewVariables(VVAccess.ReadOnly)]
+        public int samples { get; set; } = 0;
+
+        /// <summary>
+        ///     Calculate average pressure over X atmos updates.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadOnly)]
+        public int maxSamples { get; set; } = 5;
+
+        [ViewVariables(VVAccess.ReadOnly)]
+        public float[] measurements { get; set; } = new float[5];
+
+        [ViewVariables(VVAccess.ReadOnly)]
+        public float[] pressurizationRate { get; set; } = new float[5];
+
+        /// <summary>
+        ///     The vent will be shut down if the average pressure drop is less than this value.
+        /// </summary>
+        /// <remarks>
+        ///     In my testing, in a underPressureLockout failure the average pressure
+        ///     drop over 5 ticks may drop to -0.04, triggering this check and causing a lockout, solving the
+        ///     underPressureLockout failure.
+        ///
+        ///     This could be set to 0, but then the vents will stay locked for annoyingly long time whenever air flows
+        ///     from this vent to the surroundings. This may cause pressure drops in the order of E-5, causing lockouts
+        ///     while refilling rooms.
+        /// </remarks>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float pressurizationLockout { get; set; } = -0.02f;
```

Comment:
What units are these in? Please indicate that in a comment.

Reply:
Too tired to make a commit for it right now, the units are in kPa. Totally forgot that atmos works with moles a lot.

Reply:
How can the unit be in kPa? That is a pressure unit, note a pressure per time unit.

Reply:
To be honest, I'm not entirely sure what the per time part should be in this particular case, /nTicks or /s. I kinda just watched logs and picked a number until it worked. 

The moment I realized that this whole problem seems to relate to signal processing/advanced math a lot, I kinda wanted to just defenestrate my computer because I didn't study it...

Reply:
You should use time units, because tick units will differ depending on the server's tick rate and create bugs that e.g. only show up on live servers and not in development.